ec2_instance_terraform/base.tf

```diff
@@ -0,0 +1,61 @@
+provider "aws" {
+  access_key = var.aws_access_key
+  secret_key = var.aws_secret_key
+  region     = var.aws_region
+}
+
+resource "aws_iam_role" "ec2_role_test" {
+  name = "ec2_role_test"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+
+  tags = {
+      tag-key = "ec2_role_test"
+  }
+}
+
+resource "aws_iam_instance_profile" "ec2_profile_test" {
+  name = "ec2_profile_test"
+  role = aws_iam_role.ec2_role_test.name
+}
+
+resource "aws_iam_role_policy" "ec2_policy_test" {
+  name = "ec2_policy_test"
+  role = aws_iam_role.ec2_role_test.id
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:*"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_instance" "dannyinstance_1" {
+  ami           = var.aws_ami
```

Comment:
Can we use "data sources" to avoid managing ami IDs?

Reply:
Sure! thanks for the advice